### PR TITLE
publiccloud: register image before install LTP

### PR DESCRIPTION
**Pull request by asmorodskyi**
Before this patch we were trying to install LTP before registering
This make some dependecies unresolvable.
Current patch change the order - first register the system than try to install LTP

### CODE REVIEW

**Review by jlausuch — APPROVED** [review]

File: tests/publiccloud/run_ltp.pm [hunk 1]
```diff
@@ -60,6 +60,8 @@ sub run {
     assert_script_run('chmod +x restart_instance.sh');
     assert_script_run('chmod +x log_instance.sh');
 
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $REG_CODE,          timeout => 600) if (get_required_var('FLAVOR') =~ m/BYOS/);
```
Comment (jlausuch) [inline tests/publiccloud/run_ltp.pm:63]:
```suggestion
    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $REG_CODE, timeout => 600) if (get_required_var('FLAVOR') =~ m/BYOS/);
```

Comment (jlausuch) [reply tests/publiccloud/run_ltp.pm:63]: feel free to add suggestion to brach and I'll merge after green Travis

**Comment by asmorodskyi** [comment]
this is VR https://openqa.suse.de/tests/5353834 ( not done yet )

**Review by pdostal — APPROVED** [review]

File: tests/publiccloud/run_ltp.pm [hunk 2]
```diff
@@ -60,6 +60,8 @@ sub run {
     assert_script_run('chmod +x restart_instance.sh');
     assert_script_run('chmod +x log_instance.sh');
 
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $REG_CODE, timeout => 600) if (get_required_var('FLAVOR') =~ m/BYOS/);
```
Comment (cfconrad) [inline tests/publiccloud/run_ltp.pm:63]: I think we have a `is_byos()`

**Pull request merged**